Server.py

```diff
@@ -0,0 +1,157 @@
+import struct
+import select
+import socket
+import argparse
+from common import *
+
+
+class Client:
+    def __init__(self, name, room_name, address):
+        self.name = name
+        self.room_name = room_name
+        self.address = address
+
+
+class Server:
+    def __init__(self, server_port: int):
+        self.host = ''
+        self.port = server_port
+        self.address = (self.host, self.port)
+        self.main_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.sockets = []
+        self.clients = {}  # {client_socket: Client()}
+
+    def __enter__(self):
+        self.main_socket.bind(self.address)
+        self.main_socket.listen()
+        self.sockets.append(self.main_socket)
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.handle_exit()
+
+    def run(self) -> None:
+        while True:
+            readable_sockets, _, _ = select.select(self.sockets, [], [])
+            for curr_socket in readable_sockets:
+                if curr_socket == self.main_socket:
+                    new_client_socket = self.handle_new_client()
+                    print(f"New client from {self.clients[new_client_socket].address} was connected successfully!")
+                else:
+                    self.handle_message(curr_socket)
+
+    def handle_new_client(self) -> socket.socket:
+        new_socket, new_client_address = self.main_socket.accept()
+        self.sockets.append(new_socket)
+        self.clients[new_socket] = Client("", "", new_client_address)
+        return new_socket
+
+    def handle_exit(self) -> None:
+        for client_socket in self.clients.keys():
+            self.close_client(client_socket)
+        self.main_socket.close()
+
+    def handle_message(self, sender_socket: socket.socket) -> None:
+        try:
+            received_data = sender_socket.recv(RECEIVE_BUFFER_SIZE)
+        except ConnectionResetError:
+            self.close_client(sender_socket)
+            return
+
+        print(received_data)
```

Comment:
Perhaps add a debug flag for the server to decide whether to print every message